### PR TITLE
Response refactored

### PR DESCRIPTION
**Pull request by ddavlet**
Response now save and return pure data with no classes. This ensures correct data transition

Known issues:
1. Content-length causes infinite download on client side
2. Spaces in names causes incorrect parsing

other issues stated in issues

### CODE REVIEW

**Comment by ddavlet** [comment]
There is still work to do with exceptions
I will add ticket

**Review by olympiaobr — DISMISSED** [review]

**Review by olympiaobr — APPROVED** [review]

File: src/responses/Response.cpp [hunk 1]
```diff
@@ -113,46 +115,64 @@ void Response::_handleGetRequest(const Request& req) {
...
+					generateResponse(indexPath);
+				} catch (Response::FileSystemErrorException &e) {
                     _setError(404);
-                } else {
-                    setStatus(200);
-                    setBody(content);
-                    addHeader("Content-Type", _getMimeType(indexPath));
-                }
+				} catch (Response::ContentLengthException &e) {
+					throw e; //temp
```
Comment (lodeme) [inline src/responses/Response.cpp:126]: Why not return an error page here?

**Review by lodeme — COMMENTED** [review]
Can you explain the idea of pure data here?

**Review by lodeme — APPROVED** [review]
Made some minor changes. tested different use cases. works well!

**Review by olympiaobr — APPROVED** [review]

**Pull request merged**